packages/block-editor/src/components/inserter/menu.js

```diff
@@ -429,6 +456,9 @@ export default compose(
 		const {
 			getChildBlockNames,
 		} = select( 'core/blocks' );
+		const {
+			getPreference,
+		} = select( 'core/edit-post' );
```

Comment:
The inserter is a generic component from the `block-editor` package. It can't use a selector from the core/edit-post package. I wonder if the hiddenBlocks are available in the `getSettings` selector of the `block-editor` store instead.